### DISCUSSION TITLE
Should smartparens be considered troublesome?

### DISCUSSION THREAD

**Discussion by ghost** [justinbarclay/parinfer-rust-mode#93] [Ideas]
I know we can't account for _every_ character-managing plugin out there, but perhaps we should note some common ones as `troublesome` out of the box? It'd be zero-cost and save substantial headaches. smartparens is the one I think of because it's the autopairs of choice for Spacemacs _and_ DOOM, but I'm not an Emaster so I'm doubtless missing countless.

Counter-argument: knowing when a plugin manages format elements isn't hard, and accounting for every notable one would be a bigger headache for us than we'd be saving the user. Plugins can be lost track of with a distro, but the distro can set the list itself.
I don't really have a counter-counter.

**Comment by justinbarclay** [comment]
I am happy to add smartparens to the list of troublesome-modes if it improves the default experience for others. I'm curious if you have some examples of issues you've run into though? Is it when using a specific command or merely by having packages enabled?

FWIW, I'm a sadist when testing parinfer so I have smartparens and paredit on while working in Elisp. Though, I only use paredit's barf and slurp commands. I haven't noticed an issue when running when they're just turned on.

**Reply by ghost** [reply]
Personally I have not run into anything, as I use DOOM which pins a version that pulls a binary incompatible with my system (hence why this is a discussion, not an issue); I just made the inference that smartparens can autoclose pairs just like electric-pair-mode, so could be a problem for the same reason. If smartparens with autoclose actually plays just fine with parinfer, I guess this is a non-issue.

**Reply by justinbarclay** [reply]
Yeah, I think it plays well. From my experience, it's things that manage whitespace that cause parinfer issues because parinfer uses whitespace changes to infer where parens should be.

I also think that parinfer and smartparens are both smart enough to only add the trailing paren if it doesn't exist in the buffer already. At least, I assume that's why they don't cause me issues.
